test/join.jl

```diff
@@ -116,5 +116,58 @@ module TestJoin
     @test join(dt, dtnull, on = :x) ==
         DataTable([collect(1:10), collect(2:11), NullableArray(3:12)], [:x, :y, :z])
     @test join(dtnull, dt, on = :x) ==
-        DataTable([NullableArray(1:10), NullableArray(3:12), NullableArray(2:11)], [:x, :z, :y])
+        DataTable([NullableArray(1:10), NullableArray(3:12), collect(2:11)], [:x, :z, :y])
+
+    @testset "all joins" begin
+        small = DataTable(Any[[1, 3, 5], [1.0, 3.0, 5.0]], [:id, :fid])
+        large = DataTable(Any[[0, 1, 2, 3, 4], [0.0, 1.0, 2.0, 3.0, 4.0]], [:id, :fid])
+        N = Nullable()
+
+        @test join(small, large, kind=:cross) == DataTable(Any[repeat([1, 3, 5], inner=5),
```

Comment:
Break the line after `==` to avoid going beyond 92 chars.

Also, it would be good to store the result of the join and check the column types to make sure we get `NullableArray` and not `Array{Nullable}`. Should there also be tests for categorical arrays?